File: vi/codes/kotlin/utils/ListNode.kt
/**
 * File: ListNode.kt
 * Created Time: 2024-01-25
 * Author: curtishd (1023632660@qq.com)
 */

package utils

/* Nút của danh sách liên kết */
class ListNode(var _val: Int) {
    var next: ListNode? = null

    companion object {
        /* Chuyển đổi một danh sách (list) thành danh sách liên kết */
        fun arrToLinkedList(arr: IntArray): ListNode? {
            val dum = ListNode(0)
            var head = dum
            for (_val in arr) {
                head.next = ListNode(_val)
                head = head.next!!
            }
            return dum.next
        }
    }
}

File: vi/codes/python/modules/list_node.py
"""
File: list_node.py
Created Time: 2021-12-11
Author: krahets (krahets@163.com)
"""


class ListNode:
    """Lớp nút của danh sách liên kết"""

    def __init__(self, val: int):
        self.val: int = val  # Giá trị của nút
        self.next: ListNode | None = None  # Tham chiếu đến nút kế tiếp


def list_to_linked_list(arr: list[int]) -> ListNode | None:
    """Chuyển đổi một danh sách thành danh sách liên kết"""
    dum = head = ListNode(0)
    for a in arr:
        node = ListNode(a)
        head.next = node
        head = head.next
    return dum.next


def linked_list_to_list(head: ListNode | None) -> list[int]:
    """Chuyển đổi một danh sách liên kết thành danh sách"""
    arr: list[int] = []
    while head:
        arr.append(head.val)
        head = head.next
    return arr
